src/renderers/shared/fiber/ReactDebugCurrentFiber.js

```diff
@@ -34,7 +34,11 @@ function getCurrentFiberOwnerName() : string | null {
     switch (fiber.tag) {
       case IndeterminateComponent:
       case ClassComponent:
+        return getComponentName(fiber);
       case FunctionalComponent:
+        if (fiber._debugOwner != null) {
+          return getComponentName(fiber._debugOwner);
+        }
```

Comment:
If I remove this line then the following three tests fail:

```
src/renderers/shared/fiber/__tests__/React
IncrementalSideEffects-test.js
-* invokes ref callbacks after insertion/update/unmount
src/renderers/shared/shared/__tests__/ReactStatelessComponent-test.js
 * should update stateless component
 * should unmount stateless component
 * should pass context thru stateless component
-* should warn when given a string ref
-* should warn when given a function ref
```

They all fail for the same reason. This is an example of the results from the current stack reconciler.

```
    expectDev(normalizeCodeLocInfo(console.error.calls.argsFor(0)[0])).toBe(
      'Warning: Stateless function components cannot be given refs. ' +
      'Attempts to access this ref will fail. Check the render method ' +
      'of `Foo`.\n' +
      '    in FunctionalComponent (at **)\n' +
      '    in div (at **)\n' +
      '    in Foo (at **)'
    );
```

If I remove this line then Fiber ends up with:

```
      expectDev(normalizeCodeLocInfo(console.error.calls.argsFor(0)[0])).toBe(
        'Warning: Stateless function components cannot be given refs. ' +
        'Attempts to access this ref will fail. Check the render method ' +
~       'of `FunctionalComponent`.\n' +
        '    in FunctionalComponent (at **)\n' +
        '    in div (at **)\n' +
        '    in Foo (at **)'
      );
```

-----


tldr; without this line Fiber ends up logging the `FunctionalComponent`s name in the same location that stack logs the `FunctionalComponent`s **owner**s name.